network/gossip/libp2p/topology/fanout.go

```diff
@@ -0,0 +1,16 @@
+package topology
+
+import (
+	"math"
+)
+
+// FanoutFunc represents a function type that receiving total number of nodes
+// in flow system, returns fanout of individual nodes.
+type FanoutFunc func(size int) int
+
+// LinearFanoutFunc guarantees full network connectivity in a deterministic way.
+// Given system of `size` nodes, it returns `size+1/2`.
+var LinearFanoutFunc FanoutFunc = func(size int) int {
```

Comment:
The value of the fanout should actually be coming in from Scaffold as a command line argument. If none is specified then we can run with a default value of `(n+1)/2`. 
I feel like abstracting the fanout value as a function may be an overkill.

Reply:
Right, however, I faced the following two problems with the mentioned approach:
1- Fanout was getting computed slightly differently at different places of the code, which resulted in breaking the connectedness of topology in some special cases (e.g., an odd number of nodes, for example when I tried connectedness with `335` nodes, the tests broke). So I find it vital to have fanout computation abstracted at a single place. 
2- As also explained in [the other comment](https://github.com/onflow/flow-go/pull/153#discussion_r525623026), we are transiting to a phase that the fanout is an _inherent_ result of protocol, and not necessarily its input, e.g., randomized graph construction needs some probability `p` as input and constructs the graph resulting in some fanout `x` in _expectation with a high probability_ where the probability is computed over all the seeds. The entire correctness of the algorithm would be to let it have _some_ flexibility around the fanout. However, if we try to do the reverse (i.e., injecting or imposing fanout directly), the correctness of the algorithm is jeopardized as it no longer provides any probabilistic bound over the connectedness chance.